Documentation/proposals/202304-silences.md

```diff
@@ -0,0 +1,112 @@
+# Silence CRD
+
+* Owners:
+  * [mcbenjemaa](https://github.com/mcbenjemaa)
+* Related Tickets:
+  * [#5452](https://github.com/prometheus-operator/prometheus-operator/issues/5452)
+  * [#2398](https://github.com/prometheus-operator/prometheus-operator/issues/2398)
+* Other docs:
+  * n/a
+
+This document describes the creation of `Silence` Custom Resource Definition that defines Alertmanager silences
+configurations in the Kubernetes way.
+
+# Why
+
+prometheus-operator doesn't have a way to automate the management of Alertmanager silences. Many users have either been using some internal scripts
+or an additional operator that does the job e.g. [silence-operator](https://github.com/giantswarm/silence-operator).
+
+* Users who are using CI/CD jobs to manage silences have reported that this is really cumbersome.
+* Users that use scripts are not in better shape obviously.
+* Users that use a standalone operator that exposes a Silence CRD are in a better situation, some folks have reported that they are using GitOps
+  to fully manage the life cycle of Silences Custom resources, Nevertheless, this comes with a drawback
+  as the team must be able to manage a new component in their stack.
+
+Additionally, having a new component in the stack and keeping it maintained is not always ideal (said the folks at [Giant Swarm](https://giantswarm.io) the owners of [silence-operator](https://github.com/giantswarm/silence-operator)),
+because that would require us to fully manage this component as of any managed app (monitoring, logging, alerting etc..).
+Having the Silence CRD as part of prometheus-operator provides users with a better experience: there is no need to install another operator or build a system on top of it to provide that functionality.
+
+By Adding support for `Silence` CRD in the prometheus-operator, this will make users more flexible in terms of choosing the tool
+to deploy their silences and will free others from managing a standalone component within the stack.
+
+## Pitfalls of the current solution
+
+Using Alertmanager API Directly comes with drawbacks:
+
+* Teams have to build an automation to add silences in a centralized manner
+* There is no input validation, which can lead to an invalid silence configuration
+* Teams need to manage expiration on their own
+* If a silence was deleted by accident, it will be permanent.
```

Comment:
I might be talking non-sense here, but wouldn't it be possible to implement a [finalizer](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/#:~:text=Finalizers%20are%20namespaced%20keys%20that,resources%20the%20deleted%20object%20owned.) here? Making sure the silence is revoked from alertmanager before the resource gets deleted.

Reply:
No, this section explains the drawback of using the API directly. 
in the Silence CR approach, this will be fixed by the control loops that will solve the drift.